src/pyFAI/gui/dialog/DetectorSelectorDialog.py

```diff
@@ -139,18 +136,71 @@ def __init__(self, parent=None):
         self.__pixelHeight.changed.connect(self.__customDetectorChanged)
 
         self._initOrientation()
+        self._initSensor()
         # By default select all the manufacturers
         self.__selectAllRegistreredDetector()
 
     def _initOrientation(self):
-        for item in pyFAI.detectors.orientation.Orientation:
+        for item in detectors.orientation.Orientation:
             if item.available:
                 self._detectorOrientation.addItem(f"{item.value}: {item.name} ", userData=item)
 
         self._detectorOrientation.currentIndexChanged.connect(self.__orientationChanged)
-        default = pyFAI.detectors.orientation.Orientation(3)
+        default = detectors.orientation.Orientation(3)
         self._detectorOrientation.setCurrentIndex(self._detectorOrientation.findData(default))
 
+    def _initSensor(self):
+        "populate the comboBox with all available sensor materials"
+        self._sensorGroup.setEnabled(False)
+        self._detectorSensorParallax.checkStateChanged.connect(self.__sensorParallax)
+        for key, value in detectors.sensors.ALL_MATERIALS.items():
+            self._detectorSensorMaterials.addItem(key, userData=value)
+        self._detectorSensorThickness.setValidator(qt.QDoubleValidator(0.0, 1e4, 1))
+        self._detectorSensorThickness.editingFinished.connect(self.__sensorChanged)
+        self._detectorSensorMaterials.currentIndexChanged.connect(self.__sensorChanged)
+
+    def __sensorParallax(self):
+        if self._detectorSensorParallax.isChecked():
+            self._sensorGroup.setEnabled(True)
+        else:
+            self._sensorGroup.setEnabled(False)
+        self.__sensorChanged()
+
+    def getSensorConfig(self):
+        if self._detectorSensorParallax.isChecked():
+            sensor_material = self._detectorSensorMaterials.currentData()
+            sensor = detectors.sensors.SensorConfig(sensor_material)
+            thickness = self._detectorSensorThickness.text()
+            if thickness.strip():
+                sensor.thickness = 1e-6*float(thickness)
```

Comment:
Using a `QDoubleSpinBox` sparses this as well, the value is always a `float`.

Reply:
I am not sure a `QDoubleSpinBox` can be left empty to simulate an infinitely thick sensor. But my knowledge in Qt is still limited.